### PR TITLE
MAY ignore -> MUST ignore client_id, fix #143

### LINKED ISSUE #143: Value of client_id MUST be equal to origin of the redirect_uri?
Continuation of 'PS:' remark from https://github.com/remotestorage/spec/pull/141#issuecomment-165902749

I would sat 'SHOULD' not 'MUST', since you might want to ignore the rule if you implement a server for your own localhost use without this restriction.

Also, I would rephrase it from the server's perspective, so 'the server SHOULD reject OAuth requests where the `client_id` is not equal to the origin of the `redirect_uri`'.

### PR DESCRIPTION
**Pull request by michielbdejong**
With this formulation, unregistered clients can just leave out `client_id`, or set some random value which the server will ignore.

Fixes #143

### CODE REVIEW

File: source.txt [hunk 1]
```diff
@@ -486,7 +486,7 @@ Table of Contents
 
     The server MAY expire bearer tokens, and MAY require the user to
     register applications as OAuth clients before first use; if no
-    client registration is required, then the server MAY ignore the
+    client registration is required, then the server MUST ignore the
```
Comment (untitaker) [inline source.txt]: We can also remove `then` from this line.

**Comment by untitaker** [comment]
:+1:

**Comment by untitaker** [comment]
Eh, should we talk about #101 first?

**Comment by michielbdejong** [comment]
@untitaker This PR implements [your proposal](https://github.com/remotestorage/spec/issues/101#issuecomment-151885323) of "we might as well use MUST for the original text", right?

**Comment by untitaker** [comment]
Yes, but I think others disagreed

**Comment by michielbdejong** [comment]
ok, @skddc @fkooman additional review please

**Comment by fkooman** [comment]
All I wanted was to make it absolutely clear how to identify apps and to not leave any ambiguity. It seems the suggested pull request takes care of that right?

**Comment by untitaker** [comment]
@fkooman Further up that thread you seemed to vehemently argue that ignoring `client_id` in favor of `redirect_uri` (which is just a question of phrasing) would violate the OAuth spec

**Comment by fkooman** [comment]
Looks good to me!
